storage/tokudb/ha_tokudb.cc

```diff
@@ -139,30 +139,18 @@ static void free_key_and_col_info (KEY_AND_COL_INFO* kc_info) {
 }
 
 
-uchar* TOKUDB_SHARE::hash_get_key(
-    TOKUDB_SHARE* share,
-    size_t* length,
-    TOKUDB_UNUSED(my_bool not_used)) {
-
-    *length = share->_full_table_name.length();
-    return (uchar *) share->_full_table_name.c_ptr();
-}
-void TOKUDB_SHARE::hash_free_element(TOKUDB_SHARE* share) {
-    share->destroy();
-    delete share;
-}
 void TOKUDB_SHARE::static_init() {
-    my_hash_init(
-        &_open_tables,
-        table_alias_charset,
-        32,
-        0,
-        0,
-        (my_hash_get_key)hash_get_key,
-        (my_hash_free_key)hash_free_element, 0);
+    assert_always(_open_tables.size() == 0);
 }
 void TOKUDB_SHARE::static_destroy() {
-    my_hash_free(&_open_tables);
+    for (auto it = _open_tables.cbegin(); it != _open_tables.cend(); it++) {
+        TOKUDB_TRACE("_open_tables %s %p", it->first.c_str(), it->second);
+        TOKUDB_SHARE* share = it->second;
+        share->destroy();
+        delete share;
+    }
+    _open_tables.clear();
+    assert_always(_open_tables.size() == 0);
```

Comment:
I wouldn't bother with this assert right after clear call, but if it's there, it can stay